HHS/TANF-app
# 230 form validation

### This pull request changes...

This PR adds form validation to the `EditProfile` page.

**FUNCTIONALITY**
There should be no validation done until the user tries to submit the form. Upon submission, any form field that does not have a valid character (one character for first and last name and a selected STT) will receive a form group error (the line on the side of the form group), a red outline will surround the corresponding input missing its value, and a statement will appear to explain what is missing above the element.

**TO TEST**
- Go to https://tdp-frontend.app.cloud.gov/ and login
- On the login page, focus on the First Name input and then blur away from the input
    - There should be no error yet
- Repeat previous step for Last Name
    - There should be no error yet
- Repeat step for STT
    - There should be no error yet
- Click on the `Request Access` button
- All 3 inputs should have the above mentioned styling changes
- Go to `First Name` input, type in a character and blur out of it and the error messaging should go away
- Repeat for `Last Name`
- Repeat for `STT`

EDIT: Previously the card noted an outstanding issue  [370](https://github.com/raft-tech/TANF-app/issues/370). That issue has now been cleared.


### [AC] This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] Vulnerability Scan Test (Zap) has no high/critical CVEs
- [x] Linting Tests (ESLint for UI and Flake8 for Python)
- [x] Unit Tests ( including a code coverage test report)
- [x] Accessibility Tests (Pa11y UI Only)
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screen reader, text scaling) OR an exemption is documented


### [DoD] This feature is done when...
